src/axom/klee/Geometry.hpp

```diff
@@ -69,21 +78,169 @@ class Geometry
+           const Vector3D &vorDirection,
+           axom::IndexType levelOfRefinement,
+           std::shared_ptr<GeometryOperator const> operator_);
+
+  /**
+   * Create a planar Geometry object.
+   *
+   * \param startProperties the transformable properties before any
+   * operators are applied
+   * \param tet Tetrahedron
+   * \param operator_ a possibly null operator to apply to the geometry.
+   *
+   * The space on the positive normal side of the plane is considered
+   * "inside the shape".
+   */
+  Geometry(const TransformableGeometryProperties &startProperties,
+           const axom::primal::Plane<double, 3> &plane,
+           std::shared_ptr<GeometryOperator const> operator_);
+
+  /**
+   * @brief Get the format in which the geometry was specified.
+   *
+   * The format is determined by the constructor used.
+   * Values are:
+   * - "c2c" = C2C file
+   * - "proe" = ProE file
+   * - "blueprint-tets" = Blueprint tetrahedral mesh in memory
+   * - "tet3D" = 3D tetrahedron (4 points)
+   * - "sphere3D" = 3D sphere, as \c primal::Sphere<double,3>
+   * - "vor3D" = 3D volume of revolution.
+   * - "cone3D" = 3D cone, as \c primal::Cone<double,3>
+   * - "cylinder3D" = 3D cylinder, as \c primal::Cylinder<double,3>
+   * - "hex3D" = 3D hexahedron (8 points)
+   * - "plane3D" = 3D plane
    *
    * \return the format of the shape
+   *
+   * TODO: Depending on the specified geometry, some members are not
+   * used.  It may clarify make each supported geometry a subclass.
    */
   const std::string &getFormat() const { return m_format; }
 
   /**
-   * Get the path at which to find the specification of the geometry
+   * Get the path at which to find the specification of the geometry,
+   * for geometries stored in files.
    *
    * \return the path to the geometry file
    */
   const std::string &getPath() const { return m_path; }
 
-  /// Predicate that returns true when the shape has an associated geometry
-  bool hasGeometry() const { return !m_path.empty(); }
+  /**
+   * @brief Return the blueprint mesh, for formats that are specified
+   * by a blueprint mesh or have been converted to a blueprint mesh.
+   */
+  const axom::sidre::Group *getBlueprintMesh() const;
+
+  /**
+   * @brief Return the blueprint mesh topology, for formats that are specified
+   * by a blueprint mesh or have been converted to a blueprint mesh.
+   */
+  const std::string &getBlueprintTopology() const;
+
+  /**
+     @brief Return the VOR axis direction.
+  */
+  const Vector3D getVorDirection() const { return m_vorDirection; }
+
+  /**
+     @brief Return the 3D coordinates of the VOR base.
+  */
+  const Point3D getVorBaseCoords() const { return m_vorBase; }
+
+  /*! @brief Predicate that returns true when the shape has an associated geometry
+
+    A false means that this is set up to determine volume fractions without
+    computing on the geometry.
+
+    TODO: We should just create a new format to represent getting
```

Comment:
Are comments like this captured in issues so they don't get lost?

Reply:
Captured in issue https://github.com/LLNL/axom/issues/1445